src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,84 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75F;
+    private int size;
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        Entry<K, V> currentEntry = table[index];
+        while (currentEntry != null) {
+            if (Objects.equals(currentEntry.key, key)) {
+                currentEntry.value = value;
+                return;
+            }
+            currentEntry = currentEntry.next;
+        }
+        Entry<K, V> newEntry = new Entry<>(key, value);
+        newEntry.next = table[index];
+        table[index] = newEntry;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        Entry<K, V> currentEntry = table[index];
+        while (currentEntry != null) {
+            if (Objects.equals(currentEntry.key, key)) {
+                break;
+            }
+            currentEntry = currentEntry.next;
+        }
+        return currentEntry == null ? null : currentEntry.value;
```

Comment:
```suggestion
                return currentEntry.value
            }
            currentEntry = currentEntry.next;
        }
        return currentEntry;
```

Reply:
in this case last return produce NullPointerException as our table[index] may be empty (== null)